### DISCUSSION TITLE
CVE-2023-32695 doesn't affect version 4.0.2

### DISCUSSION THREAD

**Discussion by guimard** [socketio/socket.io#4721] [General] [answered]
Hi,

version 4.0.2 uses:
```typescript
return Array.isArray(payload) && typeof payload[0] === "string";
```
then this version isn't affected by this CVE.

cheers,
yadd@debian.org

**Comment by darrachequesne** [comment, marked as answer]
You are right, good catch :+1:  I've updated the version range there: https://github.com/socketio/socket.io-parser/security/advisories/GHSA-cqmj-92xf-r6r9

https://github.com/socketio/socket.io-parser/commit/1c220ddbf45ea4b44bc8dbf6f9ae245f672ba1b9 was released in version `4.0.4`.

PR for the advisory database: https://github.com/github/advisory-database/pull/2303

**Reply by darrachequesne** [reply]
The version range was updated to `>= 4.0.4, < 4.2.3`: https://github.com/advisories/GHSA-cqmj-92xf-r6r9
